File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int mx = 0 , s = 0 ;
        s = nums[0] ;
        for(int i = 1 ; i < nums.size() ; i++){
            if(nums[i] <= nums[i - 1]){
                mx = max(s , mx) ;
                s = 0 ;
            }
            s += nums[i] ;
        }
        mx = max(mx , s) ;
        return mx ;
    }
};

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int mx = 0 , s = nums[0] ;
        for(int i = 1 ; i < nums.length ; i++){
            if(nums[i] <= nums[i  - 1]){
                mx = Math.max(mx  , s) ;
                s = 0 ;
            }
            s += nums[i] ;
        }
        mx = Math.max(mx , s) ;
        return mx ;
    }
}